### ISSUE TITLE
ddl events should not be filtered in puller

### ISSUE THREAD

**Issue by CharlesCheung96** [pingcap/tiflow#10524]
The dataflow of ddl in TiCDC is:
```
TiKV --> kvClient --> ddlPuller --> owner.schemaStorage  --> owner.ddlManager  --> ddlSink
```	
There are two methods to filter ddl event:
1. Discard: filter ddl event in ddlPuller.
2. Ignore: filter ddl event in ddlManager.

In the current implementation, TiCDC discrd ddl events, which should be filtered, in [ddlPuller](https://github.com/pingcap/tiflow/blob/master/cdc/puller/ddl_puller.go#L521-L530) and the subsequent modules of puller do not handle such ddl events. This causes cdc to behave unreasonably when using  [`event-filter rules`](https://docs.pingcap.com/tidb/stable/ticdc-filter#:~:text=create%20schema,drop%20view%20event). 
For example, if we create a changefeed-test with the following config:
```toml
[filter]
rules = ['test.*']

[[filter.event-filters]]
matcher = ["test.t1"] # matcher is an allow list, which means this rule only applies to the worker table in the test database.
ignore-event = ["create table"] # Ignore `create table` ddl events.
```
Then, 
1. After executing `create table test.t0`, cdc will capture table t0.
3. After executing `create table test.t1`, cdc will discard the ddl since it match event-filter rules. Therefore, cdc will not capture table t1.
4. However, after restarting changefeed-test, cdc will get a schema snapshot from tikv, and capture both t0 and t1. This is confusing to the user.

-------
In conclusion, the behavior of TiCDC in ignoring create/delete/truncate/alter DDL in puller module is as follows:

1. After a table or partition is created, TiCDC does not synchronize the new physical table. However, cdc will synchronize the new physical table after the changefeed restarts.
5. After a table or partition is dropped, TiCDC still synchronizes the deleted physical table.
6. After a table or partition is truncated, TiCDC does not synchronize the new physical table, but still synchronizes the deleted physical table.
7. After the Alter table or partition ddl, CDC still uses the schma of the old table to parse data, wich leads to unpredictable data inconsistencies.
